### DISCUSSION TITLE
Wrong video card use on Asus Zephyrus. Likely the AMD one and not the Nvidia

### DISCUSSION THREAD

**Discussion by guysoft** [mozilla/hubs#3354] [General Questions & Discussion] [answered]
**Description**
I have a laptop that comes with an integrated GPU chip by AMD and main GPU for VR which is Nvidia RTX.
The issue is that firefox is trying to send info to the Integrated AMD GPU and not the nvidia one where the VR is connecting.

**To Reproduce**
1. Using an Asus G14, that has an AMD integrated GPU and Nvidia GPU
Connect the VR as needed using the Displayport via USB.
2. Launch Steam VE and open the developer web console to follow errors
3. Start Mozilla hubs on this link: https://hubs.mozilla.com/
4. See error "Wrong graphics card"on Steam VR
```
Tue Nov 17 2020 01:00:47.452 - Capturing Scene Focus
Tue Nov 17 2020 01:00:47.531 - Found Windows 10 or newer, so enable advanced image processing of scene textures.
Tue Nov 17 2020 01:00:47.531 - App submitted texture from wrong device!  Make sure app is configured to use the graphics card that your headset is plugged into.
Tue Nov 17 2020 01:00:47.531 - Failed to open frame info buffer (D3D10)!
Tue Nov 17 2020 01:00:47.541 - [Dismissable Warning Added] DismissableWarning_SceneApplicationUsingWrongGraphicsAdapte
```

**Expected behavior**
Firefox should send the GPU data to the Main GPU for VR

**Hardware** 
 - Device:Asus Zephyrus G14 with HTC Vive
 - OS: Windows 10
 - Browser Firefox, 82.0.3

**Additional context**
Workaround is to plug the VR using the HDMI port that is soldered to the AMD GPU. That makes hubs load, but its slow and laggy because this is an ingratiated chip and not the GPU that is built to be VR Ready.

Related information and a similar error can be seen here
Basically if you want VR to work on this laptop it needs to be on the Nvidia GPU and not the AMD which is integrated
https://forum.vive.com/topic/8693-asus-rog-zephyrus-g14-htc-vive-hdmi-issue-help/page/2/

Also other VR games seem to use the right GPU.

**Comment by misslivirose** [comment, marked as answer]
This doesn't sound like a Hubs-specific bug, but something related to how Firefox and the graphics drivers are interacting. [This Bugzilla ticket](https://bugzilla.mozilla.org/show_bug.cgi?id=1365492) seems to indicate a history of multi-GPU setups having issues. However, the most recent comment suggests that the following workaround may also work. Have you manually set the graphics card that you want an application to use in Windows 10 by going to 'Graphics Settings' and setting the browser to use your high performance GPU instead of the default / integrated card?
